include/feature_lib.h

```diff
@@ -0,0 +1,62 @@
+#ifndef FEATUREDETECT_FEATURELIB_H_
+#define FEATUREDETECT_FEATURELIB_H_
+
+#include <vector>
+#include <opencv2/features2d.hpp>
+
+namespace featurelib {
+
+struct KeyPoint {
+  float x;
+  float y;
+  float size;
+  float angle;
+  float strength;
+};
+
+struct DetectionResult {
+  std::vector<KeyPoint> keypoints;
+  std::vector<std::vector<uint8_t>> descriptors;
+};
+
+class FeatureDetector {
+ public:
+  virtual DetectionResult detect(const std::vector<std::uint8_t> &input_vector) const = 0;
+
+  void setHeight(int height) {
+    mask.reshape(1, std::vector<int>{height, width});
+  }
+
+  void setWidth(int width) {
+    mask.reshape(1, std::vector<int>{height, width});
+  }
```

Comment:
You must also update the corresponding fields in both of these setters -- otherwise what's the point?